test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -97,13 +99,66 @@ func (data *testData) createPods(t *testing.T, ns string) error {
 	return err
 }
 
+func (data *testData) assertPodNetworkStatus(t *testing.T, clientset *kubernetes.Clientset, pods []*testPodInfo, ns string) error {
+	for _, pod := range pods {
+		if err := wait.PollUntilContextTimeout(context.Background(), time.Second, 20*time.Second, false, func(ctx context.Context) (done bool, err error) {
+			podItem, err := clientset.CoreV1().Pods(ns).Get(ctx, pod.podName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
+			}
+			t.Logf("Get Pod Annotations: %+v\n", podItem.Annotations)
+
+			secondarynetworkList, _ := utils.ParsePodNetworkAnnotation(podItem)
+			t.Logf("Get Pod networks in Annotations: %+v", secondarynetworkList)
+
+			ips, ipv6, macMap, err := data.listPodIPs(pod)
+			if err != nil {
+				return false, err
+			}
+			t.Logf("Get Pod IPs: %+v\n", ips)
+
+			networkStatus, err := utils.GetNetworkStatus(podItem)
+			if err != nil || len(networkStatus) != len(ips)-2 || len(networkStatus) != len(secondarynetworkList) {
+				return false, nil
+			}
+			for _, network := range networkStatus {
+				if ip, ok := ips[network.Interface]; ok {
+					assert.Contains(t, network.IPs, ip.String(), "IPs in the Pod network status annotation should contain Pod IPs")
+				} else {
+					return false, fmt.Errorf("interface in Network status(%v) does not exist in the Pod IPs(%v), Pod: %s/%s", network, ips, ns, pod.podName)
+				}
+			}
```

Comment:
This seems a duplicate check given later it checks `expectNetworkStatuses` equals to `networkStatus`

Reply:
updated